src/vm/edge_shape.rs

```diff
@@ -129,3 +131,142 @@ fn a_simple_edge_should_have_the_same_size_as_a_pointer() {
         std::mem::size_of::<*mut libc::c_void>()
     );
 }
+
+/// A abstract memory slice represents a piece of **heap** memory.
+pub trait MemorySlice: Send + Debug + PartialEq + Eq + Clone + Hash {
+    type Edge: Edge;
+    type EdgeIterator: Iterator<Item = Self::Edge>;
+    /// Iterate _valid_ object edges within the slice.
```

Comment:
```suggestion
    /// Iterate object edges within the slice. If there are non-reference values in the slice, the iterator should skip them.
```

The sentence about the dynamic languages is also inaccurate. You should consider removing it.